### PR TITLE
FFW controller

### PR DESCRIPTION
**Pull request by gbmarc1**
Added FeedFoward network controller class
A parameters have been added to choose controller type

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 notebooks/.ipynb_checkpoints
 env
 
+.idea/*
```
Comment (loudinthecloud) [inline .gitignore:5]: This is better placed in the global .gitignore file

File: tasks/copytask.py [hunk 2]
```diff
@@ -54,6 +54,7 @@ class CopyTaskParams(object):
     name = attrib(default="copy-task")
     controller_size = attrib(default=100, convert=int)
     controller_layers = attrib(default=1,convert=int)
+    controller_type = attrib(default='LSTM', convert=str)
```
Comment (loudinthecloud) [inline tasks/copytask.py:57]: Please make it an Enum, how can I tell what are the valid options?

File: ntm/aio.py [hunk 3]
```diff
@@ -36,7 +36,14 @@ def __init__(self, num_inputs, num_outputs,
 
         # Create the NTM components
         memory = NTMMemory(N, M)
-        controller = LSTMController(num_inputs + M*num_heads, controller_size, controller_layers)
+
+        if controller_type == 'LSTM':
+            controller = LSTMController(num_inputs + M*num_heads, controller_size, controller_layers)
+        elif controller_type == 'FFW':
+            controller = FFWController(num_inputs + M * num_heads, controller_size, controller_layers)
```
Comment (loudinthecloud) [inline ntm/aio.py:43]: Please use:

`CONTROLLERS = {
    'LSTM': LSTMController,
    'FFW': FFWController
}`

Then the code has no repetitions (DRY)...

**Review by loudinthecloud — CHANGES_REQUESTED** [review]
Seems good, thanks for the contribution!

One thing that is missing is validation that it is working and converging. How does it perform?

**Comment by gbmarc1** [comment]
Thank you for the PRs. I will correct as requested. 
I will also validate the modification with both Tasks.
I am planning to add the vanilla LSTM as in the paper. I will submit all those modifications when I will be done.

**Comment by loudinthecloud** [comment]
Thanks, sounds great!

**Pull request closed without merging**